### DISCUSSION TITLE
ESP32 BLE Client is not able to find any service

### DISCUSSION THREAD

**Discussion by voed** [nanoframework/Home#1204] [Q&A]
Hi. I am running the Central1 sample code, but the service list is always empty. Of course they do have services. I can see them from the "BLE scanner" android app and even from ESPHome installed on this board.
```

=== Advert received ====
Address:A050000003
Local name:PRNAQaqPRANA            
Manufacturers Data:0
Service UUIDS:0

=== Advert received ====
Address:291E264D3B00
Local name:
Manufacturers Data:1
Service UUIDS:0

=== Advert received ====
Address:A4C138630629
Local name:
Manufacturers Data:0
Service UUIDS:0

=== Advert received ====
Address:582D3411D32D
Local name:
Manufacturers Data:0
Service UUIDS:0

=== Advert received ====
Address:A050000003
Local name:PRNAQaqPRANA            
Manufacturers Data:0
Service UUIDS:0

=== Advert received ====
Address:291E264D3B00
Local name:
Manufacturers Data:1
Service UUIDS:0

=== Advert received ====
Address:A4C138630629
Local name:
Manufacturers Data:0
Service UUIDS:0

=== Advert received ====
Address:582D3411D32D
Local name:
Manufacturers Data:0
Service UUIDS:0
```

**Comment by alberk8** [comment]
Add the watcher.ScannerMode after watcher initialize.   That should show the other details.  Most devices do not advertise the service list.  To actually get a full service list of a particular device, you need to connect to it. 

```
  BluetoothLEAdvertisementWatcher watcher = new();
  watcher.ScanningMode = BluetoothLEScanningMode.Active;
```

**Reply by voed** [reply]
Tried with active scan but result was the same. Docs says both GetGattServices and GetGattServicesForUuid will try to connect to device, but its also returns 0 services with result.Status =  0 (Success?)

**Reply by alberk8** [reply]
If you have the code on github, then I can take a look.

**Reply by voed** [reply]
@alberk8 https://github.com/voed/PranaNF/tree/test
